js/search/domain.js

```diff
@@ -515,6 +515,12 @@ const DomainV3 = new Lang.Class({
    no EKN_VERSION file is found. This function does synchronous file I/O. */
 function get_ekn_version (app_id) {
     let dir = datadir.get_data_dir(app_id);
+
+    // Sanity check
+    if (!dir) {
+        throw new Error(Format.vprintf('Could not find data dir for app ID %s', [app_id]));
```

Comment:
The `return 1;` below will actually just cause an error. It will get looked up in an `EKN_VERSIONS` map which doesn't have a key for 1, and try to construct `undefined`, which will also throw an error.